cli/src/providers/azure/InfrastructureBuilder.py

```diff
@@ -223,10 +224,19 @@ def get_storage_share_config(self):
     def get_vm(self, component_key, vm_config, availability_set, network_interface_name, security_group_association_name, index):
         vm = dict_to_objdict(deepcopy(vm_config))
         vm.specification.name = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index), component_key)
+
+        # component_key is not changed as it requires further changes in Ansible configuration
+        alt_component_name = vm.specification.alt_component_name
+        host_component_key = component_key
+        if alt_component_name and alt_component_name.strip():
+            host_component_key = alt_component_name
+
         if self.hostname_domain_extension != '':
-            vm.specification.hostname = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index) + f'.{self.hostname_domain_extension}', component_key)
+            vm.specification.hostname = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index) + f'.{self.hostname_domain_extension}', host_component_key)
         else:
-            vm.specification.hostname = vm.specification.name
+            vm.specification.hostname = resource_name(self.cluster_prefix, self.cluster_name, 'vm' + '-' + str(index), host_component_key)
+        if len(vm.specification.hostname) > 64:
+            raise Exception('Linux host name cannot exceed 64 characters in length. Consider setting alt_component_name property.')
```

Comment:
It would be a good idea to print the hostname that has exceeded the limit. Otherwise it's hard to debug and rename a component when you don't know which one.
```
12:41:36 INFO cli.src.providers.azure.InfrastructureBuilder - Starting step
12:41:36 INFO cli.src.providers.azure.InfrastructureBuilder - Step finished in: 0.14s
12:41:36 ERROR epicli - Linux host name cannot exceed 64 characters in length. Consider setting alt_component_name property.
12:41:37 INFO dump_debug_info - Error dump has been written to: /shared/epicli_error_20220405-124136.dump
```